### PR TITLE
Update Sequence and Activity Diagrams for stats

### PR DESCRIPTION
**Pull request by choongyx**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -554,6 +483,77 @@ Cell Formatting
...
+
+There are 2 types of data that the program require, `statsMessage` for displaying the statistics in text and `statsData` for displaying the data in a piechart.
+Below is the UML sequence diagram and a step-by-step explanation for the retrieval of `statsData`. `statsMessage` is retrieved in the same manner.
+
+.Sequence Diagram of Execution Phase showing retrieving of `statsData`
+image::StatsDataSequenceDiagram.png[]
+
+1. Upon checking that the boolean `isShowStats` = true, the MainWindow calls `LogicManager#getStatsData()`.
+
+2. SpendingBookParser will parse and identify the command as a StatsCommand and pass on the argument `d/ 01/01/2019 d/07/01/2019` to StatsCommandParser.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: There is a space in the input here

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -554,6 +483,77 @@ Cell Formatting
...
+
+During the preparation phase, the program parses the command for Statistics and prepares to open a new Statistics Window.
+The following sequence diagram shows how `stats d/01/01/2019 d/07/01/2019` works during the preparation phase.
+
+.Sequence Diagram demonstrating the preparation stage of Statistics feature
+image::StatsSequenceDiagram.png[]
+
+1. User executes `stats d/01/01/2019 d/07/01/2019` to generate statistics for spending within `01/01/2019` and `07/01/2019`.
+
+2. SpendingBookParser will parse and identify the command as a StatsCommand and pass on the argument `d/ 01/01/2019 d/07/01/2019` to StatsCommandParser.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: Space in input

**Review by Nanosync — CHANGES_REQUESTED** [review]
Requesting changes from previous review

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -554,6 +483,83 @@ Cell Formatting
...
+The filtering is done through a predicate parameter which checks through the dates of the spending in the application.
+
+5. Finally, the `statsData` will be displayed on the Ui through `StatsWindow#loadData()` call.
+
+==== Design Considerations
+
+===== How to pass statistics data parameters from Logic to UI
+
+====== Alternative 1 (current choice): UI gets all data and parameters from Logic, which gets data from Model.
+
```
Comment (austinsantoso) [inline docs/DeveloperGuide.adoc:548]: Are we using this format or a table format? to be consistent

Comment (Nanosync) [reply docs/DeveloperGuide.adoc:548]: Table

**Review by Nanosync — APPROVED** [review]

**Pull request merged**